src/webgpu/util/texture/texel_data.spec.ts

```diff
@@ -106,16 +107,67 @@ function doTest(
   pass.end();
   t.device.queue.submit([encoder.finish()]);
 
-  t.expectGPUBufferValuesEqual(
-    outputBuffer,
-    new ReadbackTypedArray(
-      rep.componentOrder.map(c => {
-        const value = rep.decode(componentData)[c];
-        assert(value !== undefined);
-        return value;
-      })
-    )
+  const idealReadbackData = new ReadbackTypedArray(
+    rep.componentOrder.map(c => {
+      const value = rep.decode(componentData)[c];
+      assert(value !== undefined);
+      return value;
+    })
   );
+
+  if (format === 'rgba8unorm-srgb' || format === 'bgra8unorm-srgb') {
+    // In D3D the SRGB -> Float conversion is permitted a tolerance of 0.5f ULP on the SRGB side.
+    // The procedure for measuring this tolerance is to convert the result back into SRGB space
+    // using the ideal FLOAT -> SRGB conversion specified below but WITHOUT the rounding to integer,
+    // and taking the floating point difference versus the original SRGB value to yield the error.
+    // Exact conversion is required: 0.0f and 1.0f (the ends) must be exactly achievable.
+    // https://microsoft.github.io/DirectX-Specs/d3d/archive/D3D11_3_FunctionalSpec.htm#3.2.3.7%20SRGB%20-%3E%20FLOAT
+    const readBackValue = await t.readGPUBufferRangeTyped(outputBuffer, {
+      type: Float32Array,
+      typedLength: 4,
+    });
+    t.expect(idealReadbackData[3] === readBackValue.data[3]);
+
+    // Do the ideal FLOAT -> SRGB conversion on the readback float value WITHOUT the rounding to
+    // integer.
+    const readBackValueToSRGB = new Float32Array(3);
```

Comment:
Not sure we want the intermediate round from f64 to f32...? The 0/255 cases may depend on it currently, but I think those could be simplified to simply require 0.0/1.0 specifically. Then you can use a plain array of 3 `number`s here instead.